src/analysis/rust_type.rs

```diff
@@ -91,14 +112,22 @@ fn rust_type_full(env: &Env, type_id: library::TypeId, nullable: Nullable, ref_m
 
     if type_id.ns_id != library::MAIN_NAMESPACE && type_id.ns_id != library::INTERNAL_NAMESPACE
         && !implemented_in_main_namespace(&env.library, type_id) {
-        let rust_type_with_prefix = rust_type.map(|s| format!("{}::{}",
-            crate_name(&env.library.namespace(type_id.ns_id).name), s));
         if env.type_status(&type_id.full_name(&env.library)).ignored() {
-            rust_type = Err(rust_type_with_prefix.as_str().into());
-        } else {
-            rust_type = rust_type_with_prefix;
+            rust_type = Err(rust_type.as_str().to_owned());
+        }
+        rust_type = rust_type.map_any(|s| format!("{}::{}",
+            crate_name(&env.library.namespace(type_id.ns_id).name), s));
+    }
+
+    if let Err(s) = rust_type {
```

Comment:
This is getting out of hand, I'll think about adding an enum that would also handle the "Unsupported" and "Ignored" comment.